api/account/account_api/endpoints/device.py

```diff
@@ -357,48 +316,77 @@ def patch(self, device_id: str):
         :param device_id: database identifier of a device
         """
         self._authenticate()
-        updates = self._validate_request()
-        self._update_device(device_id, updates)
+        self._validate_request()
+        self._update_device(device_id)
         self.etag_manager.expire_device_etag_by_device_id(device_id)
         self.etag_manager.expire_device_location_etag_by_device_id(device_id)
         self.etag_manager.expire_device_setting_etag_by_device_id(device_id)
 
         return "", HTTPStatus.NO_CONTENT
 
-    def _update_device(self, device_id: str, updates: dict):
+    def _validate_request(self):
+        """Validate the contents of the HTTP POST request."""
+        if self.request.method == "POST":
+            device = NewDeviceRequest(self.request.json)
+        else:
+            device = UpdateDeviceRequest(self.request.json)
+        device.validate()
+        self.validated_request = device.to_native()
+        self.validated_request.update(
+            wake_word=self.validated_request["wake_word"].lower()
+        )
+        if self.validated_request["release_channel"] is not None:
+            self.validated_request.update(
+                release_channel=self.validated_request["release_channel"].lower()
+            )
+        _log.info(self.validated_request["wake_word"])
+
+    def _ensure_geography_exists(self):
+        """If the geography does not exist in the database, add it.
```

Comment:
Does this mean people can write in cities as long as the other fields exist?

Reply:
This is the geography assigned to an account.  If the user specifies a geography for a device that is not linked to its account, a new row is added.  I will update the docstring to be more clear.